analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantInheritanceList.cs

```diff
@@ -199,14 +173,17 @@ private static bool CanInterfaceBeRemovedBasedOnMembers(INamedTypeSymbol declare
             foreach (var interfaceMember in allMembersOfInterface)
             {
                 var classMember = declaredType.FindImplementationForInterfaceMember(interfaceMember);
-                if (classMember != null
+                if (classMember is not null
                     && (classMember.ContainingType.Equals(declaredType)
-                    || !classMember.ContainingType.Interfaces.SelectMany(i => i.AllInterfaces.Concat(new[] { i })).Contains(interfaceType)))
+                    || !classMember.ContainingType.Interfaces.SelectMany(x => AllInterfacesAndSelf(x)).Contains(interfaceType)))
```

Comment:
I'm not 100% sure, but this might be a bit better
```suggestion
                    || !classMember.ContainingType.Interfaces.Any(x => AllInterfacesAndSelf(x).Contains(interfaceType))))
```